src/Features/Core/Portable/EditAndContinue/EditAndContinueDiagnosticUpdateSource.cs

```diff
@@ -48,15 +58,31 @@ public ValueTask<ImmutableArray<DiagnosticData>> GetDiagnosticsAsync(Workspace w
         /// We do not track the particular reported diagnostics here since we can just clear all of them at once.
         /// </summary>
         public void ClearDiagnostics()
-            => DiagnosticsCleared?.Invoke(this, EventArgs.Empty);
+        {
+            // No point incrementing if we are going from clean to clean
+            if (_previouslyHadDiagnostics)
+            {
+                _previouslyHadDiagnostics = false;
+                _diagnosticsVersion++;
+            }
```

Comment:
any concern about threadsafety?

Reply:
These methods are called in exactly one spot, and the caller didn't seem to worry, so I didn't worry. I'm sure @tmat can comment more though